src/Models/Nonhydrostatic2DModels/nonhydrostatic_2d_model.jl

```diff
@@ -0,0 +1,142 @@
+        (Geometry.Cartesian1Axis(), Geometry.Cartesian1Axis()),
+        @SMatrix [1.0]
+    ),)
+
+    # operators
+    # spectral horizontal operators
+    hdiv = Operators.Divergence()
+
+    # vertical FD operators with BC's
+    # interpolators
+    scalar_interp_c2f = Operators.InterpolateC2F(
+        bottom = Operators.Extrapolate(),
+        top = Operators.Extrapolate(),
+    )
+    vector_interp_c2f = Operators.InterpolateC2F(
+        bottom = Operators.SetValue(Geometry.Cartesian1Vector(0.0)),
+        top = Operators.SetValue(Geometry.Cartesian1Vector(0.0)),
+    )
+    tensor_interp_f2c = Operators.InterpolateF2C()
+
+    # gradients
+    scalar_grad_c2f = Operators.GradientC2F()
+    B = Operators.SetBoundaryOperator(
+        bottom = Operators.SetValue(Geometry.Cartesian3Vector(0.0)),
+        top = Operators.SetValue(Geometry.Cartesian3Vector(0.0)),
+    )
+
+    # divergences
+    vector_vdiv_f2c = Operators.DivergenceF2C(
+        bottom = Operators.SetValue(Geometry.Cartesian3Vector(0.0)),
+        top = Operators.SetValue(Geometry.Cartesian3Vector(0.0)),
+    )
+    tensor_vdiv_c2f = Operators.DivergenceC2F(
+        bottom = Operators.SetDivergence(Geometry.Cartesian3Vector(0.0)),
+        top = Operators.SetDivergence(Geometry.Cartesian3Vector(0.0)),
+    )
+    tensor_vdiv_f2c = Operators.DivergenceF2C(
+        bottom = Operators.SetValue(
+            Geometry.Cartesian3Vector(0.0) ⊗ Geometry.Cartesian1Vector(0.0),
+        ),
+        top = Operators.SetValue(
+            Geometry.Cartesian3Vector(0.0) ⊗ Geometry.Cartesian1Vector(0.0),
+        ),
+    )
+
+    function rhs!(dY, Y, Ya, t)
+        dYm = dY.nhm
+        dρ = dYm.ρ
+        dρuh = dYm.ρuh
+        dρw = dYm.ρw
+        dρθ = dYm.ρθ
+        Ym = Y.nhm
+        ρ = Ym.ρ
+        ρuh = Ym.ρuh
+        ρw = Ym.ρw
+        ρθ = Ym.ρθ
+
+        # advection, pressure gradients, sources
+        # density
+        @. dρ = -hdiv(ρuh)
+        @. dρ -= vector_vdiv_f2c(ρw)
+
+        # potential temperature
+        @. dρθ = -hdiv(ρuh * ρθ / ρ)
+        @. dρθ -= vector_vdiv_f2c(ρw * scalar_interp_c2f(ρθ / ρ))
+
+        # horizontal momentum
+        @. dρuh = -hdiv(ρuh ⊗ ρuh / ρ + pressure(ρθ) * I)
+        @. dρuh -= tensor_vdiv_f2c(ρw ⊗ vector_interp_c2f(ρuh / ρ))
+
+        # vertical momentum
+        # TODO: How ugly is this?! (-_-)' . ￣□￣｜｜  (ಥ﹏ಥ)
+        uh_f = @. vector_interp_c2f(ρuh / ρ)
+        @. dρw = hdiv(uh_f ⊗ ρw)
+        @. dρw += B(
+            Geometry.transform(
+                Geometry.Cartesian3Axis(),
+                -(scalar_grad_c2f(pressure(ρθ))) +
+                scalar_interp_c2f(ρ) * Geometry.Covariant3Vector(-g), # TODO!: Not generally a Covariant3Vector
+            ) - tensor_vdiv_c2f(tensor_interp_f2c(
+                ρw ⊗ ρw / scalar_interp_c2f(ρ),
+            )),
+        )
+
+        # TODO! diffusion goes here!
+
+        # discrete stiffness summation for spectral operations
+        Spaces.weighted_dss!(dρ)
```

Comment:
it's probably going to be a lot more efficient to use fields of multiple variables here

Reply:
@jakebolewski  
Agreed!
What would be the correct syntax? Does something like the following work?
```
Spaces.weighted_dss!(dYm)
```

Reply:
Yes, ideally you would aggregate all variables on their respective spaces (center, faces).  The DSS will do the right thing for all the variables on  that particular center, face field.

Reply:
Sounds good. This also propagates to the shallow water model. I open a new issue #124 and will deal with it separately.